### PR TITLE
fix: Display primaryType in signTyped instead of 'Message' string, Fixes #17076

### LINKED ISSUE #17076: primaryType is not shown during signTyped - users are signing things almost blindly
### Describe the bug

PrimaryType isn't displayed when signing Typed data, instead it always shows a "Message" string and a bunch of parameters.

IMHO this goes against the primary goal of EIP712 - a more user-friendly way of signing messaging.
It's astonishing that for 3 years people were just signing things blindly!

**Example:**
If two functions have exact same parameters but do completely different things - there is no way to distinguish between a harmless signature (like setting a limit on something), and a really dangerous one (like permitForAll that potentially can lead to loss of all your tokens).
Even if parameters differ slightly - it's still hard for a non-technical user to see what exactly they sign.

**Real world example:**
Recently there was a loss of many Lens Protocol profiles from malicious websites that asked users for a signature, mimicking a login. If user could see they are signing "permitForAll" or "burnWithSig" - they wouldn't sign it.

**Current behavior:**
telegram-cloud-photo-size-4-5870527322589870546-y

**Expected behavior:**
Before 2019 everything worked as expected, but in PR #6891 the `primaryType` was replaced with a "Message" string for a reason unknown to me (maybe @tshfx can explain more details about the reasoning of this).
The EIP712 example shows a screenshot with "Mail" as primaryType, so I assume that also confirm this behavior is the expected.

**Proposed solution:**
Display primaryType instead of "Message" (like "PermitForAll" in this case)
image

### Steps to reproduce

1. Sign any TypedData, look at the signature message window

### Error messages or log output

_No response_

### Version

10.23.2

### Build type

None

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by vicnaum**
## Explanation

Fixes #17076 

* What is the current state of things and why does it need to change?

When signing Typed data user doesn't see what primaryType (a function) they are signing, instead they always see "Message".

* What is the solution your changes offer and how does it work?

User should see the name of the function they sign (like "PermitForAll") instead of "Message" string.

### Before

image

### After

image

## Manual Testing Steps

- Sign any Typed data

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
  - [X] How this problem was solved
  - [X] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

My PR does fix this in a quick way, but I'm sure a proper fix should take more than this - would like to ask for advice for core devs here on what else should be changed, and a proposed architecture.
Also not all tests are passing on my machine (but it's the same with a clean develop branch - so maybe it's just my local issue).

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by vicnaum** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by danjm** [comment]
@vicnaum Thanks for this contribution. It looks correct and we would like to get it merged soon. There is a lint error that you will need to fix. You should be able to run `yarn verify-locales:fix` to fix the "test-lint" failure. I'll comment inline on how to fix the other failure.

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+- Added displaying the primaryType during Typed data signing ([#17077](https://github.com/MetaMask/metamask-extension/pull/17077))
```
Comment (danjm) [inline CHANGELOG.md:8]:
```suggestion
### Fixed
- Added displaying the primaryType during Typed data signing ([#17077](https://github.com/MetaMask/metamask-extension/pull/17077))
```

**Comment by danjm** [comment]
It would also be good if `ui/components/app/signature-request/signature-request.component.test.js` could be update to include an assertion for the primary type. Are you able to do that @vicnaum? If you'd prefer that we do that, please just let us know with a comment here.

**Comment by vicnaum** [comment]
@danjm Thanks! I've pushed the suggested fixes and also tried to add an assertion to the test - please check if that's what needed

File: ui/components/app/signature-request/signature-request-message/signature-request-message.js [hunk 2]
```diff
@@ -81,7 +82,7 @@ export default function SignatureRequestMessage({
           marginLeft={4}
           className="signature-request-message__title"
         >
-          {t('signatureRequest1')}
+          {primaryType}
```
Comment (jpuri) [inline ui/components/app/signature-request/signature-request-message/signature-request-message.js:85]: May be `{primaryType || t('signatureRequest1')}`.

Comment (jpuri) [reply ui/components/app/signature-request/signature-request-message/signature-request-message.js:85]: Ignore this as I found every typed message except V1 has a primary type. we are not using this component for V1.

**Review by jpuri — APPROVED** [review]

**Comment by seaona** [comment]
I am wondering what should happen when the dapp does not provide any `primaryType`. Should we handle the error somehow? At the moment, MetaMask popup tries to open but it's immediately closed, and we get an RPC error in the background console. 

See below:

https://user-images.githubusercontent.com/54408225/212046387-98fefb39-2040-402f-b22c-0a463f656f16.mp4

On the [EIP-712 specs](https://eips.ethereum.org/EIPS/eip-712), it states that both the `primaryType` and the `domain` should be mandatory, so I think it's okay what we do.

In the case of the `domain` , the same behaviour happens, and we display this different error on the console:

image

**Review by FrederikBolding — APPROVED** [review]

**Comment by holantonela** [comment]
> On the [EIP-712 specs](https://eips.ethereum.org/EIPS/eip-712), it states that both the primaryType and the domain should be mandatory, so I think it's okay what we do.

Agreed. We are also enforcing mandatory fields in other custom typesign like SIWE.

> In the case of the domain , the same behaviour happens, and we display this different error on the console

Should we unify the error message in these cases to keep consistency?

**Comment by kingdutch001** [comment]
I can confirm that this issue is still exists on latest next (v13.1.1). I use getStaticProps + getStaticPaths with fallback: blockfing and middleware. 404 pages works incorrectlry with client-side navigation. I have [slug].js pages. If I open exisiting page like /post-1 and then open 404 page like /post-not-found I see /post-1 content (what???). If I open 404 page directlry (I see 404 error page) and then open another 404 page like /post-another-not-found I see correct 404 error page. If I open some other page (not [siug].js and then open 404 page I see empty page without any content.

**Comment by brad-decker** [comment]
@FrederikBolding / @holantonela is this ready for merrge? If so one of you can do the honors :)

**Comment by seaona** [comment]
thank you for the inputs @holantonela @FrederikBolding

**Pull request merged**